.github/workflows/pypi.yml

```diff
@@ -0,0 +1,47 @@
+# This workflow will upload a Python Package using Twine when a new tag is created
+# For more information see: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#publishing-to-package-registries
+
+name: detect-secrets-pypi
+
+on:
+  push:
+    tags:
+      - v*
+
+jobs:
+
+  tox:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python: ['3.6', '3.7', '3.8', '3.9']
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python }}
+    - run: python -m pip install --upgrade setuptools pip tox virtualenv
+    # Run tox only for the installed py version on the runner as outlined in the python matrix
+    # Ensures the correct py version is installed and tested as opposed to 'tox' which attempts to run for all py versions in tox.ini
+    - run: tox -e py
+    - run: tox -e mypy
+
+  deploy:
+    # Run tests beforing deploying to pypi
+    needs: tox
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install dependencies
+      run: pip install build
+    - name: Build package
+      run: python -m build
```

Comment:
I don't think you need this `build` package, `python setup.py sdist bdist_wheel` from within the virtualenv is sufficient https://github.com/Yelp/detect-secrets/blob/4ceff679333787bdd551f2e779054553b4bbf536/scripts/upload-to-pypi#L126-L135

Reply:
Right that was my inital approach. I received this and thought it might be best to use the `build`package. 
`setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(`

Reply:
`build` on this repo is basically just going to call `setup.py sdist bdist_wheel` anyway, if you want to keep it though, can you add it to `requirements-dev[-minimal].txt`?